### PR TITLE
Update static display and fit

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: es6/jsxcad-ui-threejs.js [hunk 1]
```diff
@@ -1375,38 +1382,48 @@ const moveToFit = ({
...
+    new Vector3(box.min.x, box.max.y, box.min.z),
+    new Vector3(box.max.x, box.max.y, box.min.z),
+    new Vector3(box.min.x, box.min.y, box.max.z),
+    new Vector3(box.max.x, box.min.y, box.max.z),
+    new Vector3(box.min.x, box.max.y, box.max.z),
+    new Vector3(box.max.x, box.max.y, box.max.z),
+  ];
 
-  camera.position.copy(center).sub(direction);
+  // Back the camera off by 1mm at a time until the bounding box is contained by
```
Comment (BarbourSmith) [inline es6/jsxcad-ui-threejs.js:1398]: That's neat!

File: es6/jsxcad-ui-threejs.js [hunk 2]
```diff
@@ -1375,38 +1382,48 @@ const moveToFit = ({
...
+  for (let distance = 1; ; distance += 1) {
+    const direction = target
+      .clone()
+      .sub(camera.position)
+      .normalize()
+      .multiplyScalar(distance);
+
+    camera.position.copy(target).sub(direction);
+
+    camera.near = distance / 100;
```
Comment (BarbourSmith) [inline es6/jsxcad-ui-threejs.js:1410]: I was seeing an issue where if I created an orbitView() with a small object in it and then updated it with a larger object the near and far planes would be wrong so part of the object could get cut off. Does this run just once when the orbitView is created, or every time it is updated?

**Review by BarbourSmith — APPROVED** [review]
Some of these seem less in frame than before. It seems like specifically flat shapes maybe? 3D stuff seems to look great.

image

File: es6/jsxcad-ui-threejs.js (hunk 2 above)
Comment (pentacular) [reply es6/jsxcad-ui-threejs.js:1410]: Just when it's created -- the camera should be persistent through updates, so that you can see changes in-place.

**Comment by pentacular** [comment]
> Some of these seem less in frame than before. It seems like specifically flat shapes maybe? 3D stuff seems to look great.
> 
> image

Ah, maybe it's not contributing to the bounding box properly -- let's see.

**Comment by pentacular** [comment]
> > Some of these seem less in frame than before. It seems like specifically flat shapes maybe? 3D stuff seems to look great.
> > image
> 
> Ah, maybe it's not contributing to the bounding box properly -- let's see.

Good catch -- the world matrix wasn't being updated so the bounding box was taking the unscaled shape.

File: es6/jsxcad-ui-threejs.js (hunk 2 above)
Comment (BarbourSmith) [reply es6/jsxcad-ui-threejs.js:1411]: Could we have a way to update just the near and far distances? I'm finding situations where I create a view of something small and then it grows as I work on it. Eventually parts of the geometry stick out past the far plane and aren't rendered anymore

Comment (BarbourSmith) [reply es6/jsxcad-ui-threejs.js:1411]: That would probably do it. I have made things bigger than 10m (house model).

Comment (pentacular) [reply es6/jsxcad-ui-threejs.js:1411]: Let's try 1 km, and see how that goes. :)

**Pull request merged**